nano/secure/ledger.hpp

```diff
@@ -53,9 +53,16 @@ class ledger final
 	bool block_exists (store::transaction const & transaction, nano::block_hash const & hash) const;
 	nano::uint128_t account_balance (store::transaction const &, nano::account const &, bool = false);
 	nano::uint128_t account_receivable (store::transaction const &, nano::account const &, bool = false);
+	/**
+	 * Returns the cached vote weight for the given representative.
+	 * If the weight is below the cache limit it returns 0.
+	 * During bootstrap it returns the preconfigured bootstrap weights.
+	 */
 	nano::uint128_t weight (nano::account const &);
 	std::optional<nano::block_hash> successor (store::transaction const &, nano::qualified_root const &) const noexcept;
 	std::optional<nano::block_hash> successor (store::transaction const & transaction, nano::block_hash const & hash) const noexcept;
+	/* Returns the exact vote weight for the given representative by doing a database lookup */
+	nano::uint128_t weight_exact (store::transaction const &, nano::account const &);
```

Comment:
On the naming of "weight_exact" it seems like the other weight function isn't exact though they should always be in sync shouldn't they?

Reply:
Let's say a representative has a vote weight of Ӿ0.1. `weight_exact` would return Ӿ0.1, but `weight` would return 0, because it wasn't in the cache. I wanted to rename `weight` to `weight_cached`, but this would pollute this pull request with the renames. Or should I rename it now? Do you have better names in mind?

Reply:
Renaming is going to impact work in progress code, the current "non-exact" weight function is what node components should use 99% of the time.